### PR TITLE
buildkite-cli: fix shebangs

### PR DESCRIPTION
**Pull request by wpcarro**
While awaiting this PR: https://github.com/buildkite/cli/pull/124

TIL `patchShebangs` exists (ty @sternenseemann)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/continuous-integration/buildkite-cli/default.nix [hunk 1]
```diff
@@ -15,6 +15,14 @@ buildGoModule rec {
 
   doCheck = false;
 
+  preBuild = ''
+    files=(
+      .buildkite/steps/lint.sh
+      .buildkite/steps/run-local.sh
+    )
+    patchShebangs "''${files[@]}"
```
Comment (Artturin) [inline pkgs/development/tools/continuous-integration/buildkite-cli/default.nix:23]:
```suggestion
    patchShebangs ".buildkite/steps/{lint.sh,run-local.sh}"
```

**Comment by Artturin** [comment]
Please follow our commit msg format described in contributing.md

File: pkgs/development/tools/continuous-integration/buildkite-cli/default.nix [hunk 2]
```diff
@@ -15,6 +15,14 @@ buildGoModule rec {
 
   doCheck = false;
 
+  preBuild = ''
```
Comment (SuperSandro2000) [inline pkgs/development/tools/continuous-integration/buildkite-cli/default.nix:18]: This should be postPatch if the files are not generated in configurePhase.

**Comment by wpcarro** [comment]
Thanks for the feedback :)

File: pkgs/development/tools/continuous-integration/buildkite-cli/default.nix [hunk 3]
```diff
@@ -15,6 +15,10 @@ buildGoModule rec {
 
   doCheck = false;
 
+  postPatch = ''
+    patchShebangs ".buildkite/steps/{lint,run-local}.sh"
```
Comment (SuperSandro2000) [inline pkgs/development/tools/continuous-integration/buildkite-cli/default.nix:19]:
```suggestion
    patchShebangs .buildkite/steps/{lint,run-local}.sh
```

**Pull request merged**